### ISSUE TITLE
Multi-GPU training bug

### ISSUE THREAD

**Issue by likeoff** [minimaxir/textgenrnn#151]
I trying to train same model with all same hardware/software/hyperparameters using single GPU and Multi GPU.

After training at single GPU, I find that everything works as described, but after training with 3 GPUs, textgenrnn can't generate text - it prints only prefix and no other words / characters.

model_cfg = {
    'word_level': False, 
    'rnn_size': 512, 
    'rnn_layers': 1,  
    'rnn_bidirectional': False,  
    'max_length': 550,  
    'max_words': 10000, 
}

train_function(
    multi_gpu=False,
    file_path=file_name,
    new_model=True,
    num_epochs=train_cfg['num_epochs'],
    gen_epochs=train_cfg['gen_epochs'],
    batch_size=200,
    train_size=train_cfg['train_size'],
    dropout=train_cfg['dropout'],
    validation=train_cfg['validation'],
    is_csv=train_cfg['is_csv'],
    rnn_layers=model_cfg['rnn_layers'],
    rnn_size=model_cfg['rnn_size'],
    rnn_bidirectional=model_cfg['rnn_bidirectional'],
    max_length=model_cfg['max_length'],
    dim_embeddings=100,
    word_level=model_cfg['word_level'],
)


Here is output with model trained on 3 GPUs:

textgen = textgenrnn(config_path="Motherboards-512x1-multi-gpu_config.json", weights_path="Motherboards-512x1-multi-gpu_weights.hdf5",
vocab_path="Motherboards-512x1-multi-gpu_vocab.json")
textgen.generate(prefix = 'intel', max_gen_length=1048, temperature=0.9, n=3 )

 33%|███▎      | 1/3 [00:55<01:51, 55.88s/it]

intel                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  

 67%|██████▋   | 2/3 [01:51<00:55, 55.86s/it]

intel                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  

100%|██████████| 3/3 [02:47<00:00, 55.87s/it]

intel

**Comment by likeoff** [comment]
Same with single GPU, 512-LSTM cells, single layer.

**Comment by likeoff** [comment]
Same with 256-LSTM, 2 layers. I think sometimes model just do not want to educate.

**Comment by breadbrowser** [comment]
it only uses cpu
